### PR TITLE
fixes i18n routing #10963

### PR DESCRIPTION
**Pull request by chrabyrd**
### Types of changes

-   [x] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Currently in dev/7.6.x, if multiple languages are enabled at the project level, building a static asset bundle will succeed, but the template for `language-switcher.htm` will fall back to the django error template. This is because in [this PR](https://github.com/archesproject/arches/pull/10754), the line 

```
urlpatterns.append(path("i18n/", include("django.conf.urls.i18n")))
```

has been moved to the project level, and commented out by default. Without including the Django i18n urls, the form action in `language-switcher.htm` ( `{% url 'set_language' %}` ) will fail loudly as Django isn't able to find the url. 

Those lines were commented out by default because of [this issue](https://github.com/archesproject/arches/issues/8831) . The changes in [this PR](https://github.com/archesproject/arches/pull/10754) fix said issue.

This change un-comments those lines in project-level `urls.py`, adds the interaction to arches-core `urls.py`, and adds future-proofing for when urls using Django's `i18n_patterns` are included from Arches applications.  This also hardens the webpack logic to loudly fail if the server consistently responds with a 500

### Issues Solved

#10963 

### Checklist

-   [ ] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: arches/install/arches-templates/project_name/urls.py-tpl [hunk 1]
```diff
@@ -7,6 +10,11 @@ urlpatterns = [
...
 ] + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
 
-# if settings.SHOW_LANGUAGE_SWITCH is True:
-#     urlpatterns = i18n_patterns(*urlpatterns)
-#     urlpatterns.append(path("i18n/", include("django.conf.urls.i18n")))
\ No newline at end of file
+if settings.SHOW_LANGUAGE_SWITCH is True:
+    # Only handle i18n routing in active project. This will still handle the routes provided by Arches core and Arches applications,
+    # but handling i18n routes in multiple places causes application errors.
+    directory_path = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
```
Comment (jacobtylerwalls) [inline arches/install/arches-templates/project_name/urls.py-tpl:16]: Thanks for jumping on this. I feel like we run into subtle issues with editable installs and virtualization and symlinks often enough that I'm wondering if we can skin this apple another way, maybe like:
```py
if settings.APP_NAME != "Arches" and settings.APP_NAME not in settings.ARCHES_APPLICATIONS
```
Would that be equivalent?

File: arches/install/arches-templates/project_name/urls.py-tpl [hunk 2]
```diff
@@ -1,3 +1,6 @@
+import os
+import inspect
+
```
Comment (jacobtylerwalls) [inline arches/install/arches-templates/project_name/urls.py-tpl:3]:
```suggestion
```

File: arches/urls.py [hunk 3]
```diff
@@ -15,11 +15,12 @@
 You should have received a copy of the GNU Affero General Public License
 along with this program. If not, see <http://www.gnu.org/licenses/>.
 """
+import os
+import inspect
```
Comment (jacobtylerwalls) [inline arches/urls.py:19]:
```suggestion
```

File: releases/7.6.0.md [hunk 4]
```diff
@@ -234,15 +234,32 @@ Minor incompatibilities:
...
-        *Note: If the previous code was uncommented in your project make sure to uncomment the new code.*
-        ```
-        # if settings.SHOW_LANGUAGE_SWITCH is True:
-        #     urlpatterns = i18n_patterns(*urlpatterns)
-        #     urlpatterns.append(path("i18n/", include("django.conf.urls.i18n")))
-        ```
+9. Update urls.py:
+    ```
+    import os
+    import inspect
```
Comment (jacobtylerwalls) [inline releases/7.6.0.md:241]: needs update

File: releases/7.6.0.md [hunk 5]
```diff
@@ -234,15 +234,32 @@ Minor incompatibilities:
...
+
+    urlpatterns = [
+        path('', include('arches.urls')),
+        ...  # other project-level routes
+    ] + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+
+    if settings.SHOW_LANGUAGE_SWITCH is True:
+        # Only handle i18n routing in active project. This will still handle the routes provided by Arches core and Arches applications,
+        # but handling i18n routes in multiple places causes application errors.
+        directory_path = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
```
Comment (jacobtylerwalls) [inline releases/7.6.0.md:257]: needs update

File: arches/urls.py [hunk 6]
```diff
@@ -314,6 +314,13 @@
     re_path(r"^transform-edtf-for-tile", api.TransformEdtfForTile.as_view(), name="transform_edtf_for_tile"),
 ]
 
+if settings.SHOW_LANGUAGE_SWITCH is True:
```
Comment (jacobtylerwalls) [inline arches/urls.py:317]: At the project level, when this is False, I get:

```py
django.urls.exceptions.NoReverseMatch: Reverse for 'set_language' not found. 'set_language' is not a valid view function or pattern name.
[29/May/2024 09:21:01] "GET /templates/views/components/language-switcher.htm HTTP/1.1" 500 134864
```

File: arches/app/views/main.py [hunk 7]
```diff
@@ -62,11 +62,7 @@ def function_templates(request, template):
 def templates(request, template):
     if not template:
         template = request.GET.get("template")
-
-    try:
-        return render(request, template)
-    except Exception as e:
-        print(e)
+    return render(request, template)
```
Comment (jacobtylerwalls) [inline arches/app/views/main.py:65]: 🚀

**Review by jacobtylerwalls — COMMENTED** [review]
Noticed a 500 in the webpack build, and I think it's new as of this PR, but I also know that this PR surfaces more errors in general.

**Review by apeters — APPROVED** [review]
looks good!

**Comment by chrabyrd** [comment]
converting to draft until the urls.py issue is in 👍

**Review by jacobtylerwalls — APPROVED** [review]
Tested with both values for SHOW_LANGUAGE_SWITCH at project level. lgtm, thanks!

**Pull request merged**